lms/templates/widgets/footer-language-selector.html

```diff
@@ -34,23 +34,30 @@
 </div>
 
 <script type="text/javascript">
-    ##
-    ## Set the language cookie using the same settings as
-    ## https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/views.py#L26
-    ## and
-    ## https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/middleware.py#63
-    ##
-    function setLanguage($select) {
-        var cookie = '${settings.LANGUAGE_COOKIE | n, js_escaped_string}=' + $select.value + ';path=/';
-        % if settings.SESSION_COOKIE_DOMAIN:
-            cookie += ';domain=${settings.SESSION_COOKIE_DOMAIN | n, js_escaped_string}';
-        % endif
-        % if COOKIE_DURATION:
-            cookie += ';max-age=${COOKIE_DURATION | n, js_escaped_string}';
-        % endif
+    window.footerLanguageSelector = {
+        handleSelection: function($select) {
+            this.setLanguageCookie($select.value);
+            this.refreshPage();
```

Comment:
I think it would be safer to pass this as a callback to the ```setLanguageCookie``` just in case page refresh is called before the cookie is set.